bspump/elasticsearch/connection.py

```diff
@@ -607,9 +600,31 @@ def build_headers(username, password, api_key):
 			'Content-Type': 'application/json',
 			"Authorization": 'ApiKey {}'.format(api_key)
 		}
-		# L.log(asab.LOG_NOTICE, 'Building headers with api_key')
 	else:
 		auth = None
 		Headers = None
 
 	return Headers, auth
+
+def get_credentials_from_config(configuration):
+	"""
+	Parses configurations to return credentials
+	"""
+
+	# new configuration
+	username = configuration.get('username')
+	password = configuration.get('password')
+	api_key = configuration.get('api_key')
+	url = configuration.get('url')
+
+	# backward compatibility with the format http://<username>:<password>@path
+	auth_credentials = configuration.get('elasticsearch_url')
+	if auth_credentials is not None:
+		asab.LogObsolete.warning("The configuration of ESConnection is outdated, please update the [connection:] section.")
+		pattern = re.compile(r"(https?://)((.*):(.*)@)?([^#]*)$")
+		path_split = pattern.findall(auth_credentials)[0]
+		L.debug(path_split)
+		url_scheme, user_info, username, password, url_path = path_split
```

Comment:
You are getting these information from `Config["elasticsearch_url"]`. But what if someone specifies it in `Config["url"]`?

Reply:
In this case can I please get a better input on old config options please? Isn't it this format for "old" configuration format: http://<username>:<password>@path? Also, Config["url"] is handled in the first part of the function or I probably don't get what you mean?

Reply:
I bet the "elasticsearch_url" option has never been in bspump (it is an option in asab:storage section) and it was only typo in the task description. Elena, please use real site configs as a reference if you're not sure.